### PR TITLE
docs(storage): clean up JS v6 storage doc

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes:
Clean up the storage doc for v6 launch:
* Clear confusing guidance of users setting up the external S3 bucket the Auth by themselves. None of other libraries includes it.
* Add missing imports in code examples
* Re-organize the sections of `downloadData()`

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/lib/storage/getting-started/q/platform/[platform].mdx [hunk 1]
```diff
@@ -81,225 +99,10 @@ import js0 from '/src/fragments/lib/common/js/import_configuration.mdx'
 ```typescript
 import { Amplify } from 'aws-amplify';
 import awsconfig from './aws-exports';
```
Comment (ashwinkumar6) [inline src/pages/lib/storage/getting-started/q/platform/[platform].mdx:101]:
```suggestion
import { Amplify } from 'aws-amplify';
import amplifyconfig from './amplifyconfiguration.json';
```

File: src/pages/lib/storage/getting-started/q/platform/[platform].mdx [hunk 2]
```diff
@@ -81,225 +99,10 @@ import js0 from '/src/fragments/lib/common/js/import_configuration.mdx'
...
-      "x-amz-server-side-encryption",
-      "x-amz-request-id",
-      "x-amz-id-2",
-      "ETag",
-      "x-amz-meta-foo" 
-    ],
-    "MaxAgeSeconds": 3000
-  }
-]
+Amplify.configure(awsconfig);
```
Comment (ashwinkumar6) [inline src/pages/lib/storage/getting-started/q/platform/[platform].mdx:103]:
```suggestion
Amplify.configure(amplifyconfig);
```

**Review by ashwinkumar6 — APPROVED** [review]
lgtm

File: src/pages/lib/storage/upload/q/platform/[platform].mdx [hunk 3]
```diff
@@ -52,7 +52,7 @@ try {
       onProgress, // Optional progress callback.
     },
   }).result;
-  console.log('Key from Response: ', result.key);
+  console.log('Succeed: ', result);
```
Comment (renebrandel) [inline src/pages/lib/storage/upload/q/platform/[platform].mdx:55]:
```suggestion
  console.log('Succeeded: ', result);
```

Can we apply the same copy change below?

**Review by renebrandel — APPROVED** [review]

File: src/pages/lib/storage/upload/q/platform/[platform].mdx [hunk 4]
```diff
@@ -69,7 +69,7 @@ try {
     key: filename,
     data: file,
   }).result;
-  console.log('Key from Response: ', result.key);
+  console.log('Succeed: ', result);
```
Comment (renebrandel) [inline src/pages/lib/storage/upload/q/platform/[platform].mdx:72]:
```suggestion
  console.log('Succeeded: ', result);
```

File: src/pages/lib/storage/upload/q/platform/[platform].mdx [hunk 5]
```diff
@@ -88,7 +88,7 @@ try {
       accessLevel: 'protected',
     },
   }).result;
-  console.log('Key from Response: ', result.key);
+  console.log('Succeed: ', result);
```
Comment (renebrandel) [inline src/pages/lib/storage/upload/q/platform/[platform].mdx:91]:
```suggestion
  console.log('Succeeded: ', result);
```

File: src/pages/lib/storage/upload/q/platform/[platform].mdx [hunk 6]
```diff
@@ -107,7 +107,7 @@ try {
       accessLevel: 'private'
     },
   }).result;
-  console.log('Key from Response: ', result.key);
+  console.log('Succeed: ', result);
```
Comment (renebrandel) [inline src/pages/lib/storage/upload/q/platform/[platform].mdx:110]:
```suggestion
  console.log('Succeeded: ', result);
```

**Review by renebrandel — APPROVED** [review]

**Pull request merged**